src/restruct_poc/constraints/calculate_cylindrical_constraint.hpp

```diff
@@ -101,8 +110,12 @@ struct CalculateCylindricalConstraint {
         //---------------------------------
         {
             // Extract gradient block for target node of this constraint
-            const auto B =
-                Kokkos::subview(gradient_terms, i_constraint, Kokkos::ALL, cd.target_node_col_range);
+            const auto B = Kokkos::subview(
+                gradient_terms, i_constraint, Kokkos::ALL,
+                Kokkos::make_pair(
+                    node_col_range(i_constraint, 1, 0), node_col_range(i_constraint, 1, 1)
```

Comment:
Again, this change significantly reduces readability of the code. Where is it documented that `node_col_range(i_constraint, 1, 0)` refers to the target node and `node_col_range(i_constraint, 0, 0)` refers to the base node? It's not clear from scanning through this code. I don't see how calling `make_pair` every time this is run is more efficient than calculating it up front and giving it a descriptive name.

Reply:
I'll look into cleaning this up.  I'll split it into `base_node_col_range` and `target_node_col_range` and use pairs in them.

I chose to go to local `make_pair` because it is very cheap (just copying 2 `size_t` from heap memory to stack memory), and this makes the load properly coalesced on GPU (the entries are further away from each other).

Reply:
Yes, using `base_node_col_range` and `target_node_col_range` makes it much easier to read the code and understand the intent.